### PR TITLE
Correctly handle twitch subscriber emotes

### PR DESCRIPTION
**Pull request by Nogesma**
This PR switches to parsing the emotes irc tag to find twitch emotes to display, to actually correctly display emotes from people that are subbed to a channel.
Users subscribed to a channel can use emotes from this channel into other channels chat, this enables detection of such emotes.

There's still a bit of work to do to detect emotes that the current user can use, to display them in the emote picker, and I need to clean up the code and do some tests, so this is why this is a draft PR.

This also reverts the change that added the '-animated' string at the end of animated twitch emotes (#587), as I found it complicates stuff a bit. (ie. I would need to check the cache for both the animated and normal version if they had different name). The only issue is that users would need to delete the cache to switch to the animated version of the twitch emotes, if they had the static one already in cache.
If you would prefer to not need user intervention, I can try to think of a way to avoid this, but clearing the cache is definitely the easier option.

### CODE REVIEW

**Comment by Nogesma** [comment]
Seems like rust 1.78 added a bunch of clippy lints. How do you want to proceed with those?

**Comment by Xithrius** [comment]
You can ignore them if you want. As long as it builds and actually functions, all is good.

**Comment by Xithrius** [comment]
> If you would prefer to not need user intervention, I can try to think of a way to avoid this, but clearing the cache is definitely the easier option.

Clearing the cache sounds good to me.

**Comment by Nogesma** [comment]
Okay this should be feature complete now, feel free to test. 
I am not subscribed to any twitch channel so I'm not really able to test the emote picker and sending messages that contains subscriber only emotes. But receiving them works nicely on my end.

I ended up moving the code around a bit in the message parsing, as the message can be now parsed 3 different ways depending on where it came from. (This is mostly why the diff is so large)
- If it came from twitch, then it needs to be parsed using both the emotes provided in the irc tags, and the global emotes.
- If it came from the current user, it needs to be parsed using the user emotes, and the global emotes.
- When the emotes finished downloading, the previous messages will have already been parsed with the twitch irc tag emotes, but not with the global ones, so they will need to be reparsed with that.

I also now cache the twitch user id when the first request is made, so getting the following list should be a bit quicker now.

I still need to document this stuff somewhere and maybe cleanup a few things, but this should be ready to merge soon.

**Comment by Nogesma** [comment]
This should now be ready to merge once #595 is merged. I rebased on top of this PR so that checks pass.

**Comment by Nogesma** [comment]
Just rebased on top of main

**Comment by Nogesma** [comment]
I had committed my cargo.toml by mistake, fixed that

**Comment by Nogesma** [comment]
I fixed a few issues I found while testing this some more. Hopefully everything should be good now. Also rebased on top of main.

**Comment by Nogesma** [comment]
Oops broke the tests, should be good now.

**Comment by Xithrius** [comment]
Sorry I haven't been able to review this, been focusing on a couple other projects. Hope to review in the next week or so.

**Review by Xithrius — APPROVED** [review]
Ran it after clearing cache, works perfectly. Just making sure, should I still warn users about a non-cleared cache in the release, or is that unnecessary? Thanks!

**Comment by Nogesma** [comment]
Yes, clearing cache is still needed, but not clearing it will just make twt use the static images for already downloaded animated twitch emotes.
You should probably also mention the `user:read:emotes` scope that is needed to display the subscriber only-emotes that someone can use.

Thanks for the review!

**Comment by Nogesma** [comment]
Just rebased on top of main

**Comment by Xithrius** [comment]
Thank you again!

**Comment by Xithrius** [comment]
I'll put this in a release once #611 is resolved.

**Comment by Xithrius** [comment]
Available in https://github.com/Xithrius/twitch-tui/releases/tag/v2.6.11

**Pull request merged**
